modules/dtj/Animal/lib/Cow.pm

```diff
@@ -4,9 +4,12 @@ use 5.006;
 use strict;
 use warnings;
 
-use parent qw(Animal);
+use Moose;
+use namespace::autoclean;
 
-sub sound { 'moooo' }
+has 'sound' => ( is => 'ro', default => 'moooo' );
+
+with 'Animal';
```

Comment:
`has` has to be above `with` otherwise `with` bails out because there is no `sound`.